### DISCUSSION TITLE
collaborate on a python activity pub implementation

### DISCUSSION THREAD

**Discussion by georgkrause** [bookwyrm-social/bookwyrm#2461] [Ideas]
Hi there Book people!

I am Georg currently spending most of my time at Funkwhale and a user and Fan of Bookwurm. So first things first: Thank you for the great tool and your standing and all the work all the contributors did! Much love <3

At Funkwhale we are currently working towards breaking down the whole thing into smaller bits and pieces in order to improve maintainability. Major efforts go into building up a UI Library which we can use in all our projects.

I don't know if this is a road you want to walk, but it came to my mind it might be a good idea to split away the AP implementation to allow others to easily use it and to work collaborative on things. It might even enable devs to easily add AP support to their applications or start creating a new project. 

Obviously there is a lot to discuss to make it work, but I for now wanted to ask if you are even interested in those kinds of collaborations. Would you use such a library? Would you spend time to build it? Give me your feedback!

**Comment by mouse-reeve** [comment]
I'm super interested in this! I still need to wrap my head around how an ActivityPub library would work, but I'm definitely interested in doing that and discussing further. My capacity isn't super high right now, but I think it's a worthwhile endeavor to at least investigate :)

**Reply by georgkrause** [reply]
Sounds good!

I don't have a clue about how to actually build this, but it should be interesting to learn. As a first step I'd check out some other implementations, the only one I know of right now would be https://github.com/go-fed/activity/

Another thing we could do is read each others implementation so we can look for common patterns. I guess mechanics might be close while we use totally different Activities.

**Reply by georgkrause** [reply]
@mouse-reeve FYI: We got some funding for this, so there will be some progress for sure. If you are interested to do paid work here, let me know :)

**Reply by mouse-reeve** [reply]
Oof, I think the delay in me replying to this is as good an indication as any that I just don't have capacity for more paid work. I can ask a couple contributors who I think would have the know-how, if that's helpful!

**Reply by georgkrause** [reply]
@mouse-reeve Don't worry, its alright :) If there is someone wanting to collaborate on this, I'd appreciate this. At least for some loose discussions it would be nice to have someone, but I can take over the task itself if required

**Reply by Thovi98** [reply]
Just passing by, after discovering Bookwyrm (it’s awesome! 🤩).

@SamR1, I take the liberty to ping you here, as you might be interested as well for [FitTrackee](https://github.com/SamR1/FitTrackee)
